Submitted By: Joe Ciccone <jciccone@gmail.com>
Date: 2010-11-21
Initial Package Version: 1.17.3
Upstream Status: From Upstream
Origin: http://busybox.net/downloads/fixes-1.17.3/
Description: This patch includes the following upstream patches.

             busybox-1.17.3-dnsd.patch
             busybox-1.17.3-iproute.patch
             busybox-1.17.3-ls.patch
             busybox-1.17.3-sort.patch
             busybox-1.17.3-unicode.patch

diff -Naur busybox-1.17.3.orig/coreutils/ls.c busybox-1.17.3/coreutils/ls.c
--- busybox-1.17.3.orig/coreutils/ls.c	2010-10-09 15:57:13.000000000 -0400
+++ busybox-1.17.3/coreutils/ls.c	2010-11-21 08:38:13.024346417 -0500
@@ -184,7 +184,7 @@
 	LIST_INO,                   /* i */
 	LIST_LONG | STYLE_LONG,     /* l - remember LS_DISP_HR in mask! */
 	LIST_SHORT | STYLE_SINGLE,  /* 1 */
-	0,                          /* g (don't show group) - handled via OPT_g */
+	0,                          /* g (don't show owner) - handled via OPT_g */
 	LIST_ID_NUMERIC,            /* n */
 	LIST_BLOCKS,                /* s */
 	DISP_ROWS,                  /* x */
@@ -621,7 +621,7 @@
 	if (all_fmt & LIST_ID_NAME) {
 		if (option_mask32 & OPT_g) {
 			column += printf("%-8.8s ",
-				get_cached_username(dn->dstat.st_uid));
+				get_cached_groupname(dn->dstat.st_gid));
 		} else {
 			column += printf("%-8.8s %-8.8s ",
 				get_cached_username(dn->dstat.st_uid),
@@ -631,7 +631,7 @@
 #endif
 	if (all_fmt & LIST_ID_NUMERIC) {
 		if (option_mask32 & OPT_g)
-			column += printf("%-8u ", (int) dn->dstat.st_uid);
+			column += printf("%-8u ", (int) dn->dstat.st_gid);
 		else
 			column += printf("%-8u %-8u ",
 					(int) dn->dstat.st_uid,
diff -Naur busybox-1.17.3.orig/coreutils/sort.c busybox-1.17.3/coreutils/sort.c
--- busybox-1.17.3.orig/coreutils/sort.c	2010-10-09 15:57:13.000000000 -0400
+++ busybox-1.17.3/coreutils/sort.c	2010-11-21 08:38:13.024346417 -0500
@@ -412,7 +412,7 @@
 #if ENABLE_FEATURE_SORT_BIG
 	/* Open output file _after_ we read all input ones */
 	if (option_mask32 & FLAG_o)
-		xmove_fd(xopen3(str_o, O_WRONLY, 0666), STDOUT_FILENO);
+		xmove_fd(xopen3(str_o, O_WRONLY|O_CREAT|O_TRUNC, 0666), STDOUT_FILENO);
 #endif
 	flag = (option_mask32 & FLAG_z) ? '\0' : '\n';
 	for (i = 0; i < linecount; i++)
diff -Naur busybox-1.17.3.orig/libbb/unicode.c busybox-1.17.3/libbb/unicode.c
--- busybox-1.17.3.orig/libbb/unicode.c	2010-10-09 15:58:13.000000000 -0400
+++ busybox-1.17.3/libbb/unicode.c	2010-11-21 08:38:13.028345948 -0500
@@ -1006,7 +1006,7 @@
 			}
 		}
 		if (stats)
-			stats->byte_count = stats->unicode_count = (d - dst);
+			stats->byte_count = stats->unicode_count = stats->unicode_width = (d - dst);
 		return dst;
 	}
 
diff -Naur busybox-1.17.3.orig/networking/dnsd.c busybox-1.17.3/networking/dnsd.c
--- busybox-1.17.3.orig/networking/dnsd.c	2010-10-09 15:57:14.000000000 -0400
+++ busybox-1.17.3/networking/dnsd.c	2010-11-21 08:38:13.020345600 -0500
@@ -388,7 +388,7 @@
 	query_len = strlen(query_string) + 1;
 	/* may be unaligned! */
 	unaligned_type_class = (void *)(query_string + query_len);
-	query_len += sizeof(unaligned_type_class);
+	query_len += sizeof(*unaligned_type_class);
 	/* where to append answer block */
 	answb = (void *)(unaligned_type_class + 1);
 
diff -Naur busybox-1.17.3.orig/networking/libiproute/iproute.c busybox-1.17.3/networking/libiproute/iproute.c
--- busybox-1.17.3.orig/networking/libiproute/iproute.c	2010-10-09 15:57:14.000000000 -0400
+++ busybox-1.17.3/networking/libiproute/iproute.c	2010-11-21 08:38:13.020345600 -0500
@@ -791,8 +791,8 @@
 				}
 				req.r.rtm_dst_len = addr.bitlen;
 			}
-			argv++;
 		}
+		argv++;
 	}
 
 	if (req.r.rtm_dst_len == 0) {
